app/controllers/projects_controller.rb

```diff
@@ -82,6 +82,18 @@ def deploy_group_versions
     render json: deploy_group_versions
   end
 
+  def find_via_repository_url
+    repository_url = params.require(:url)
+    projects = Project.where(
+      Project.arel_table[:repository_url].matches("%#{ActiveRecord::Base.send(:sanitize_sql_like, repository_url)}%")
```

Comment:
ah nvm missed the first `%` ... but needs some docs on why this extra logic is here ...
so basically this is `find_by_user_repo_part` or so ... not meant to be used with a url

Reply:
... current logic in index will need some love though and use a common scope and not 3 different scopes :D

Reply:
see deploys_controller#search for an example ... we can pair on that tomorrow since that's all a little funky ... or grab ryan :)

Reply:
Yes, we can pair, the reason I didn't use "index" action because it totally depended on listing projects for the particular user.
As this method is specially for API, so that it should able to fetch or filter all projects.
hmmm, we can discuss more tomorrow :)